frontend/packages/console-shared/src/components/dashboard/utilization-card/prometheus-hook.ts

```diff
@@ -35,4 +35,39 @@ export const usePrometheusQuery: UsePrometheusQuery = (query, humanize) => {
   return results;
 };
 
+export const usePrometheusQueries: UsePrometheusQueries = (queries, humanize) => {
+  const dispatch = useDispatch();
+  React.useEffect(() => {
+    queries.forEach((query) => dispatch(watchPrometheusQuery(query)));
+    return () => {
+      queries.forEach((query) => dispatch(stopWatchPrometheusQuery(query)));
+    };
+  }, [dispatch, queries]);
+
+  const queryResults = useSelector<RootState, ImmutableMap<string, any>>(({ dashboards }) => {
```

Comment:
It looks to me like this will cause unnecessary re-rendering in components.

This `useSelector` will return a new object each time it runs even if no data has changed. That in turn will trigger a new `results` object to be created and returned from the hook, which will likely cause the component to re-render.

Also, whenever a single query runs, a new object will be returned, likely causing a full re-render of all values (perhaps even if no value actually changed).

If a component has several queries, it's probably easier for that component to manage its performance with several separate `usePrometheusQuery` calls rather than a single `usePrometheusQueries` call.

Maybe this is acceptable? Or maybe I'm missing something?

Reply:
Yeah, I realized this after using it in one of my components. I have made a few changes around the mechanism of how we are selecting PTAL.